mongodb_video_player/scripts/video_server.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python
+
+import rospy
+import cv2
+from sensor_msgs.msg import Image
+from cv_bridge import CvBridge
+from mongodb_media_server import MediaClient
+from mongodb_video_player.srv import VideoPlayerService, VideoPlayerServiceRequest, VideoPlayerServiceResponse
+
+from os.path import join, exists, expanduser
+from os import makedirs
+
+from threading import Thread
+
+
+class VideoServer(object):
+    def __init__(self, name):
+        rospy.loginfo("Creating "+name+" server")
+        self.video_folder = join(expanduser('~'), '.ros', 'video_server')
+        self.video_set = rospy.get_param('~video_set', 'video_set')
+        self.video_pub = rospy.Publisher(
+            rospy.get_param("~video_pub", "~video"),
+            Image,
+            queue_size=2
+        )
+        self.video_index = 0
+        self.rate = rospy.Rate(25)
+        self.bridge = CvBridge()
+        self.thread = None
+
+        self.mc = MediaClient(
+            rospy.get_param("mongodb_host"),
+            rospy.get_param("mongodb_port")
+        )
+
+        sets = self.mc.get_sets("Video")
+        object_id = None
+        for s in sets:
+            if s[0] == self.video_set:
+                object_id = s[2]
+
+        if object_id is None:
+            rospy.logwarn('Could not find any set in database matching video_set')
+            return
+
+        file_set = self.mc.get_set(object_id)
+
+        if len(file_set) == 0:
+            rospy.logwarn('No videos available in set ' + self.video_set)
+
+        for f in file_set:
+            print "Media name:", f[0]
+
+        if not exists(self.video_folder):
+            makedirs(self.video_folder)
+
+        for f in file_set:
+            file = self.mc.get_media(str(f[2]))
+            outfile = open(join(self.video_folder, f[0]), 'wb')
+            filestr = file.read()
+            outfile.write(filestr)
+            outfile.close()
+
+        self.file_names = [join(self.video_folder, f[0]) for f in file_set]
+
+        self.service = rospy.Service('video_player_service', VideoPlayerService, self.pressed_button)
+        rospy.loginfo(" ... started "+name)
+
+    def pub_video(self, filename):
+        try:
+            video = cv2.VideoCapture(filename)
+        except:
+            rospy.logwarn("Video %s not found" % filename)
+            return
+        if not video.isOpened():
+            rospy.logwarn("Capture stream could not be opened")
+            return
+
+        ret, frame = video.read()
+        while self.play and ret and not rospy.is_shutdown():
+            img = self.bridge.cv2_to_imgmsg(frame, "bgr8")
+            self.video_pub.publish(img)
+            self.rate.sleep()
```

Comment:
so, this assumes always be 25Hz videos. Quite a bold assumption. Can the rate be read from the video file?